File: pdam-backend/pdam.api/src/main/java/com/salesianostriana/dam/pdam/api/post/dto/NewPostDto.java
package com.salesianostriana.dam.pdam.api.post.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;
import org.hibernate.validator.constraints.URL;

import javax.validation.constraints.NotEmpty;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class NewPostDto {

    @Length(max = 25, message = "{newPostDto.affair.max}")
    private String affair;

    @NotEmpty(message = "{newPostDto.content.notEmpty}")
    @Length(max = 250, message = "{newPostDto.content.max}")
    private String content;

}

File: pdam-frontend/pdam_app/lib/models/post/NewPostDto.dart
class NewPostDto {
  late String? affair;
  late String content;

  NewPostDto({this.affair, required this.content});

  NewPostDto.fromJson(Map<String, dynamic> json) {
    affair = json['affair'];
    content = json['content'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['affair'] = this.affair;
    data['content'] = this.content;
    return data;
  }
}
